barman/utils.py

```diff
@@ -902,3 +902,39 @@ def lock_files_cleanup(lock_dir, lock_directory_cleanup):
                 _logger.debug(
                     "%s file lock already acquired, skipping removal" % filename
                 )
+
+
+def edit_config(file, section, option, value):
+    """
+    Utility method that given a file and a config section allows to:
+        - add a new section if at least a key-value content is provided
+        - add a new key-value to a config section
+        - change a section value
+
+    :param file: the path to the file to edit
+    :param section: the config section to edit or to add
+    :param option: the config key to edit or add
+    :param value: the value for the config key to update or add
+    """
+    conf_section = False
+    idx = 0
+    with open(file, "r") as config:
+        lines = config.readlines()
+        eof = len(lines) - 1
+        for idx, line in enumerate(lines):
+            if conf_section and line.startswith("["):  # next section
```

Comment:
```suggestion
            if conf_section and line.strip().startswith("["):  # next section
```

`ConfigParser` doesn't care about spaces around the brackets:

```python
>>> from io import StringIO
>>> from configparser import ConfigParser
>>> parser = ConfigParser(strict=False)
>>> string = StringIO("""
...         [barman]
... some_option = some_value
...
... """)
>>> parser.read_file(string)
>>> parser.sections()
['barman']
```